library/apisupport.py

```diff
@@ -28,60 +29,41 @@ def read_last_emails(email: str, creds: str, count = None) -> list[dict]:
             g.close()
 
     @staticmethod
-    def write_to_kafka_cal(events: dict) -> None:
-        producer = KafkaProducer(bootstrap_servers=os.getenv('KAFKA_BROKER','127.0.0.1:9092'), 
-                                 api_version="7.3.2", 
-                                 value_serializer=lambda v: json.dumps(v).encode('utf-8'))
-        count = 0
-        
-        for event in events:
-            count += 1
-            if event == None:
-                print("There are no events")
-                continue
-            producer.send('calendar', value = event)
+    def write_emails_to_kafka(emails: list[dict]) -> None:
+        APISupport.write_to_kafka(emails, 'emails',  lambda item: str(item['to'][0]))
 
-    
-        producer.flush()
-        print("Wrote " + str(count) + " calendar events to Kafka")
+    @staticmethod
+    def write_slack_to_kafka(slacks: list[dict]) -> None:
+        APISupport.write_to_kafka(slacks, 'slack', lambda item: str(item['name']))
 
+    @staticmethod
+    def write_cal_to_kafka(events: list[dict]) -> None:
+        APISupport.write_to_kafka(events, 'calendar')
+
+    @staticmethod
+    def write_docs_to_kafka(doc_info: list[dict]) -> None:
+        APISupport.write_to_kafka(doc_info, 'documents')
```

Comment:
I see how you've abstracted the method. There were parts of the code that were repetitive unnecessarily.